### PR TITLE
Feature/devsu 1570 add source column to kb matches

### PR DESCRIPTION
**Pull request by Anna-log7**
This is for adding 2 new columns to the kb-matches tables: externalSource and externalStatementId. No reports have these fields yet but I added tests to simulate the fields being filled out

Additionally I cleaned up and split out the new tab link/ensembl link renderer

### CODE REVIEW

File: app/components/DataTable/components/CivicCellRenderer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+const CivicCellRenderer = ({
+  data,
+}: CivicCellRendererProps): JSX.Element => {
+  const [link, setLink] = useState('');
+  const [text, setText] = useState('');
+
+  useEffect(() => {
+    if (data?.externalSource?.toLowerCase() === 'civic'
+      && typeof data?.externalStatementId === 'number') {
```
Comment (kttkjl) [inline app/components/DataTable/components/CivicCellRenderer/index.tsx:16]: would it help to coerce this into a number? or is this one of the fields that can be string as well

**Review by kttkjl — COMMENTED** [review]
Just some questions, else components look to be doing what it is supposed to 

For the future: Storybook would help a bit in displaying some of these mocked up tests

**Comment by Anna-log7** [comment]
> 
> 
> Just some questions, else components look to be doing what it is supposed to
> 
> For the future: Storybook would help a bit in displaying some of these mocked up tests

Interesting. I didn't realize Storybook had testing tools built into it. I'll make a ticket to investigate adding it to the project

File: app/components/DataTable/components/CivicCellRenderer/__tests__/CivicCellRenderer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/components/DataTable/components/CivicCellRenderer/__tests__/CivicCellRenderer.test.tsx:1]: It should yes. Looks like the `tsconfig.json` file isn't including test files though. I'll fix that.

File: app/views/ReportView/components/KbMatches/columnDefs.ts [hunk 3]
```diff
@@ -164,6 +164,16 @@ const columnDefs = [{
...
   hide: true,
+}, {
+  headerName: 'External Source',
+  colId: 'externalSource',
+  cellRenderer: 'CivicCellRenderer',
+  hide: false,
+}, {
+  headerName: 'External Statement ID',
+  colId: 'externalStatementId',
+  field: 'externalStatementId',
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/views/ReportView/components/KbMatches/columnDefs.ts:175]: It's the other way around - field takes care of `colId`. You're right though that `colId` isn't necessary here though so I'll remove it

File: app/components/DataTable/components/CivicCellRenderer/index.tsx (hunk 1 above)
Comment (Anna-log7) [reply app/components/DataTable/components/CivicCellRenderer/index.tsx:16]: Good catch. The API actually has this set as `Sq.TEXT` so I'll try using `parseInt` on it and verify with `isNan`

File: app/components/NewTabLink/index.scss [hunk 4]
```diff
@@ -0,0 +1,8 @@
+@import '../../styles/variables';
+
+.new-tab-link {
+  color: $underline;
+  font-weight: 700;
+  text-decoration: underline solid $underline;
+  cursor: pointer;
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/components/NewTabLink/index.scss:7]: Nope, that's unnecessary and I'll remove it

**Review by kttkjl — APPROVED** [review]

**Pull request merged**